### PR TITLE
Use filepath.clean instead of path.clean

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
closes https://github.com/ethereum/go-ethereum/issues/22367

NOT TESTED UNDER WINDOWS

### CODE REVIEW

**Review by rjl493456442 — APPROVED** [review]

**Comment by holiman** [comment]
Failing test:
```
    test_cmd.go:261: (stderr:21) INFO [01-02|15:39:51.806] IPC endpoint opened                      url=\\.\pipe\\pipe\geth621383
```
i.e
```
url=\\.\pipe\\pipe\geth621383
```
```
  --- FAIL: TestAttachWelcome/ipc (3.01s)
        run_test.go:107: endpoint \\.\pipe\geth621383 did not open within 3
```

**Comment by fjl** [comment]
Named pipe paths are not file paths, they exist in a separate namespace on windows and should not be expanded by this function. Probably best to add a special case for paths starting with `\.pipe`

**Review by karalabe — DISMISSED** [review]
Plop

File: internal/flags/flags_test.go [hunk 1]
```diff
@@ -19,21 +19,38 @@ package flags
...
+	if runtime.GOOS == "windows" {
+		tests = map[string]string{
+			"/home/someuser/tmp":              "\\home\\someuser\\tmp",
+			"~/tmp":                           user.HomeDir + "\\tmp",
+			"~thisOtherUser/b/":               "~thisOtherUser\\b",
+			"$DDDXXX/a/b":                     "\\tmp\\a\\b",
+			"/a/b/":                           "\\a\\b",
+			"C:\\Documents\\Newsletters\\":    "C:\\Documents\\Newsletters\\",
+			"C:\\":                            "C:\\",
+			"\\\\.\\pipe\\\\pipe\\geth621383": "\\\\.\\pipe\\\\pipe\\geth621383",
```
Comment (holiman) [inline internal/flags/flags_test.go:39]: This (by which I mean all of them) would be better expressed with backticked strings, IMO
```suggestion
			`\\.\pipe\\pipe\geth621383`: `\\.\pipe\\pipe\geth621383`,
```

File: internal/flags/flags.go [hunk 2]
```diff
@@ -314,12 +314,16 @@ func GlobalBig(ctx *cli.Context, name string) *big.Int {
 // 3. cleans the path, e.g. /a/b/../c -> /a/c
 // Note, it has limitations, e.g. ~someuser/tmp will not be expanded
 func expandPath(p string) string {
+	// Named pipes are not file paths on windows, ignore
+	if strings.HasPrefix(p, "\\\\.\\pipe") {
```
Comment (holiman) [inline internal/flags/flags.go:318]:
```suggestion
	if strings.HasPrefix(p, `\\.\pipe`) {
```

File: internal/flags/flags_test.go [hunk 3]
```diff
@@ -19,18 +19,38 @@ package flags
...
+		}
+	} else {
+		tests = map[string]string{
+			"/home/someuser/tmp":              "/home/someuser/tmp",
+			"~/tmp":                           user.HomeDir + "/tmp",
+			"~thisOtherUser/b/":               "~thisOtherUser/b",
+			"$DDDXXX/a/b":                     "/tmp/a/b",
+			"/a/b/":                           "/a/b",
+			"C:\\Documents\\Newsletters\\":    "C:\\Documents\\Newsletters\\",
```
Comment (holiman) [inline internal/flags/flags_test.go:48]: Is this correct? Shouldn't the expansion use `/` path separators? 
```suggestion
			"C:\\Documents\\Newsletters\\":    "C:\\Documents\\Newsletters\\",
```

**Pull request merged**